src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToArrayTransformer.php

```diff
@@ -167,12 +167,12 @@ public function reverseTransform($value)
         try {
             $dateTime = new \DateTime(sprintf(
                 '%s-%s-%s %s:%s:%s',
-                empty($value['year']) ? $this->referenceDate->format('Y') : $value['year'],
-                empty($value['month']) ? $this->referenceDate->format('m') : $value['month'],
-                empty($value['day']) ? $this->referenceDate->format('d') : $value['day'],
-                empty($value['hour']) ? $this->referenceDate->format('H') : $value['hour'],
-                empty($value['minute']) ? $this->referenceDate->format('i') : $value['minute'],
-                empty($value['second']) ? $this->referenceDate->format('s') : $value['second']
+                $value['year'] ?? $this->referenceDate->format('Y'),
+                $value['month'] ?? $this->referenceDate->format('m'),
+                $value['day'] ?? $this->referenceDate->format('d'),
```

Comment:
For day and month, zero doesn't mean anything, so that we should keep using the fallback when this happens. For BC at least. I think for year too, even if year zero exists.

Reply:
Fair point, I'll change those back